File: 백준/Bronze/1157. 단어 공부/단어 공부.py
import sys

a = list(sys.stdin.readline().strip().upper())

count = {}

for i in a:
    if i not in count:
        count[i] = 1
    else:
        count[i] += 1

cnt = 0

for j, k in count.items():
    if k == max(count.values()):
        cnt += 1

if cnt >= 2:
    print('?')
else:
    print(max(count, key=count.get))

File: 백준/Bronze/1157. 단어 공부/단어 공부.js
const rl = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout,
});

rl.question('', (line) => {
    let chars = line.toUpperCase().split('');
    let count = {};
    let result = '';

    for (let char of chars) {
        if (char in count) {
            count[char] += 1;
        } else {
            count[char] = 1;
        }
    }

    const values = Object.values(count);
    const max = Math.max(...values);
    const maxchar = Object.keys(count).filter((a) => count[a] === max);

    if (maxchar.length > 1) {
        result = '?';
    } else {
        result = maxchar[0];
    }

    console.log(result);
    rl.close();
});
